### PR TITLE
move dev dependencies out of requirements.txt

### LINKED ISSUE #27: [FEATURE] Don't install testing dependencies when installing via pip
* **What are you trying to do?**
The list of dependencies that is installed together with Jake at the moment is extremely long. Not all of these requirements are needed for running Jake, but only for its development. 

* **What feature or behavior is this required for?**
Having a lightweight environment to run Jake from, e.g. as part of a CI process.

* **How could we solve this issue? (Not knowing is okay!)**
Revising the list of dependencies installed via `install_requires`, only including the minimum set necessary for running Jake. Other dependencies can be moved to `tests_requires`, if they are necessary to run the tests.

* **Anything else?**
As a follow-up step, it might make sense to relax the defined versions, to make Jake play nicer with other projects installed in the same repo.

cc @bhamail / @DarthHater

### PR DESCRIPTION
**Pull request by jwa5426**
Moves any dependencies not needed for the actual package (i.e. for testing, code formatting, etc) into the dev requirements.

This pull request makes the following changes:
* Moves any dependencies not necessary for `install_requires` from `requirements.txt` into `requirements-dev.txt`
* Updates `Pipfile` and `Pipfile.lock` to reflect this ^
* Adds instruction to pip install from `dev-dependencies.txt` when running tests locally

It relates to the following issue #s:
* Fixes #27

cc @bhamail / @DarthHater

### CODE REVIEW

**Comment by jwa5426** [comment]
I’m a bit confused by the CI error, could someone point me in the right direction?

File: .github/CONTRIBUTORS.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Community Contributors
+
+- Jacob A. - https://github.com/jwa5426
```
Comment (bhamail) [inline .github/CONTRIBUTORS.md:3]: 👍

File: .circleci/ci-setup.sh [hunk 2]
```diff
@@ -3,14 +3,28 @@ set -e
...
+PATH=$WORKSPACE/.venv_non_dev/bin:$PATH
 python3 --version
-if [ ! -d ".venv" ]; then
+if [ ! -d ".venv_non_dev" ]; then
         # use python3 to create .venv
-        python3 -m venv .venv
+        python3 -m venv .venv_non_dev
 fi
-source .venv/bin/activate
+source .venv_non_dev/bin/activate
```
Comment (bhamail) [inline .circleci/ci-setup.sh:13]: So sorry you ran into this issue. Makes me think we really need to investigate moving to `poetry` or something that does a nicer job of differentiating runtime vs development dependency sets.

Eventually, we'll probably have to jump through similar flaming hoops for the internal build (see: Jenkinsfile) when a non-dev environment is needed during the CI build to self-scan.

**Review by bhamail — APPROVED** [review]
Looks good to me. Thanks!

I may wait a little before merging so others can chime in. (Please ping me if too much time elapses).

**Comment by jwa5426** [comment]
@bhamail Feel free to keep this open as long as you need, just commenting so this doesn't get forgotten.

**Comment by bhamail** [comment]
@jwa5426 Thanks for the PR!

**Pull request merged**